contracts/GNTDeposit.sol

```diff
@@ -126,39 +128,40 @@ contract GNTDeposit is ERC223ReceivingContract {
         balances[msg.sender] = 0;
         locked_until[msg.sender] = 0;
         require(token.transfer(_to, _amount));
-        Withdraw(msg.sender, _to, _amount); // event
+        Withdraw(msg.sender, _to, _amount);
     }
 
     function burn(address _whom, uint256 _burn)
         onlyOracle
         external
-        returns (bool)
     {
-        require(balances[_whom] >= _burn);
-        balances[_whom] -= _burn;
-        if (balances[_whom] == 0) {
-            locked_until[_whom] = 0;
-        }
-        require(token.transfer(0xdeadbeef, _burn));
-        Burn(_whom, _burn); // event
-        return true;
+        _reimburse(_whom, 0xdeadbeef, _burn);
+        Burn(_whom, _burn);
     }
 
-    function reimburse(address _owner, address _receiver, uint256 _reimbursement)
+    function reimburseForSubtask(address _requestor, address _provider, uint256 _amount, string _subtask_id)
```

Comment:
Are you sure about type string? In other places bytes are used. Moreover, subtask id has fixed size as far as I can recall.

Reply:
I can use bytes. But I'm not so sure whether it should be fixed size. While we are indeed using just uuid at the moment I believe there's a chance it may change. I wouldn't restrict ourselves to fixed size here.